plugins/app/src/main/java/com/mapbox/mapboxsdk/plugins/testapp/activity/GeoJsonActivity.java

```diff
@@ -0,0 +1,250 @@
+                        @Override
+                        public void onPreLoading() {
+                            progressBar.setVisibility(View.VISIBLE);
+                        }
+
+                        @Override
+                        public void onLoaded() {
+                            Toast.makeText(GeoJsonActivity.this, "GeoJson data from url loaded", Toast.LENGTH_LONG).show();
+                            progressBar.setVisibility(View.INVISIBLE);
+                        }
+
+                        @Override
+                        public void onLoadFailed(Exception e) {
+                            progressBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(GeoJsonActivity.this, "Error occur during load GeoJson data from url. see logcat", Toast.LENGTH_LONG).show();
+                            e.printStackTrace();
+                        }
+                    })
+                    .withMarkerClickListener(new OnMarkerEventListener() {
+                        @Override
+                        public void onMarkerClickListener(Marker marker, JSONObject properties) {
+                            try {
+                                Toast.makeText(GeoJsonActivity.this, properties.getString("title"), Toast.LENGTH_LONG).show();
+                            } catch (JSONException e) {
+                                e.printStackTrace();
+                            }
+                        }
+                    })
+                    .draw();
+        }
+    }
+
+    @OnClick(R.id.fabAssets)
+    public void onAssetsFabClick() {
+        if (mapboxMap != null) {
+            mapboxMap.clear();
+            new GeoJsonPluginBuilder()
+                    .withContext(this)
+                    .withMap(mapboxMap)
+                    .withFileNameAssets("IRN.geo.json")
+                    .withOnLoadingFileAssets(new OnLoadingGeoJsonListener() {
+                        @Override
+                        public void onPreLoading() {
+                            progressBar.setVisibility(View.VISIBLE);
+                        }
+
+                        @Override
+                        public void onLoaded() {
+                            Toast.makeText(GeoJsonActivity.this, "GeoJson data from assets loaded", Toast.LENGTH_LONG).show();
+                            progressBar.setVisibility(View.INVISIBLE);
+                        }
+
+                        @Override
+                        public void onLoadFailed(Exception e) {
+                            progressBar.setVisibility(View.INVISIBLE);
+                            Toast.makeText(GeoJsonActivity.this, "Error occur during load GeoJson data from assets. see logcat", Toast.LENGTH_LONG).show();
+                            e.printStackTrace();
+                        }
+                    })
+                    .draw();
+            requestExternalStoragePermission();
```

Comment:
may I use https://github.com/afollestad/material-dialogs#file-selector-dialogs for selecting GeoJson file from external storage?